src/software/ai/hl/stp/tactic/dribble/dribble_fsm.cpp

```diff
@@ -106,6 +136,33 @@ void DribbleFSM::dribble(const Update &event)
             event.control_params.dribble_destination,
             event.control_params.final_dribble_orientation);
 
+    Point dribble_destination = getDribbleBallDestination(
+        event.common.world.ball().position(), event.control_params.dribble_destination);
+
+    // when we have the ball in the dribbler, only then we pivot around the ball
+    // if (distance(event.common.robot.position(), event.common.world.ball().position()) <
+    // 0.088)
+    if (event.common.robot.isNearDribbler(event.common.world.ball().position()))
+    {
+        Angle ang_1 = event.common.robot.orientation() + Angle::fromDegrees(40);
```

Comment:
why 40 degrees here, and 30 degrees above?

Reply:
So rotation while `dribbl`ing  has much lower radius than rotation while in `getPossession`, This means that we can rotate with larger intervals in `getPosssession` because we would not be moving the ball from its position. But when we are in `dribble`, pivoting with larger angles makes the ball deviates from its position. 

I can make them both 30, only downside would be that we pivot more slowly. lmk what would be preferred.

Reply:
I think we'd prefer faster pivoting, so I think what you have is fine
but maybe abstract out the common code to an abstract function